library/core/src/num/f32.rs

```diff
@@ -1282,15 +1282,14 @@ impl f32 {
     #[must_use = "method returns a new number and does not mutate the original value"]
     #[stable(feature = "clamp", since = "1.50.0")]
     #[inline]
-    pub fn clamp(self, min: f32, max: f32) -> f32 {
+    pub fn clamp(mut self, min: f32, max: f32) -> f32 {
         assert!(min <= max);
-        let mut x = self;
-        if x < min {
-            x = min;
+        if self <= min {
```

Comment:
> however, it seems to produce this code according to godbolt

I think if you change it back to `<` instead of `<=`
```suggestion
        if self < min {
```
it'll be happy again <https://godbolt.org/z/vcz388zT1>

Reply:
(This actually matters for things like signed zeros, and though it's certainly unfortunate that `.clamp(0.0, 1.0)` can return a negative zero, we probably can't change that now.)